### ISSUE TITLE
Test fail with python-3.10

### ISSUE THREAD

**Issue by mikelolasagasti** [dalibo/pg_activity#194]
I packaged pg_activity for [Fedora](https://src.fedoraproject.org/rpms/pg_activity) and as future Fedora-35 will have python-3.10 some tests are being done already.

tests fail with python-3.10 as reported [downstream](https://bugzilla.redhat.com/show_bug.cgi?id=1937228) and also with [travis](https://github.com/dalibo/pg_activity/pull/193)

**Comment by tirkarthi** [comment]
It seems repr order of flags changed in https://github.com/python/cpython/pull/24215 to follow definition order. 

Python 3.8 : `<Flag.APPNAME|DATABASE: 3>`
Python 3.10 :  `<Flag.DATABASE|APPNAME: 3>`

**Comment by dlax** [comment]
Failure of `pgactivity/types.py::pgactivity.types.Deserializable` appears to be due to a change in python-attrs about the `type` attribute of attr-classes' fields:

python3.10:
```python
>>> @attr.s(auto_attribs=True)
... class X:
...   y: int
... 
>>> attr.fields(X)[0].type
'int'
```

python3.7:
```python
>>> @attr.s(auto_attribs=True)
... class X:
...    y: int
...
>>> attr.fields(X)[0].type
<class 'int'>
```
* * *
In any case, test failures seems to be due to assumptions made in tests which are now invalid; so I suppose the program would still work as expected. I'll look into fixing tests, though.

Thanks for the report and for your packaging and QA effort!

**Comment by dlax** [comment]
All tests are fixed on Python 3.10 now, and a version 2.1.2 just got released.
